adr/0021-rules-for-entity-mapping-in-service-catalog.md

```diff
@@ -0,0 +1,175 @@
+* **Deck Component of Prow**
+
+In special cases of complicated architectures, where there are multiple entities in between a `Domain`, `System` and a deployment, the `Component` resource can describe a "System-like" resources. The most obvious example here is applications which belong to Open Data Hub:
+
+> Open Data Hub is described as a `System` resource in Data Science `Domain`. Due to lack of intermediate resource types in Backstage, all the applications within Open Data Hub, like Superset, JupyterHub, Argo Workflows, etc. are described as `Component` resources. These resources have subcomponents (specified as `Component` resources with the [`.spec.subcomponentOf` field](https://backstage.io/docs/features/software-catalog/descriptor-format#specsubcomponentof-optional) specified) which finally map to individual deployments
+
+* **JupyterHub Component**
+* **JupyterHub Controller subcomponent** `Component` resource
+* **Idle culler subcomponent** `Component` resource
+
+The `Component` resource requires a [`.spec.type` field](https://backstage.io/docs/features/software-catalog/descriptor-format#spectype-required) to be specified. While any arbitrary value is allowed here, we recognize following main types:
+
+* `service`: Any backend-like application, API provider, bot
+* `website`: Has to provide frontend web interface
+* `library`: Provides a library or any sort of code base dependency
+* `template`: Is used as a repository or service template
+
+### Kind: API
+
+If a component is a backend service, it may expose an API intended to be consumed by others. Each exposed API should provide an definition, either statically or dynamically. Backstage [requires us to specify](https://backstage.io/docs/features/software-catalog/descriptor-format#kind-api) `.spec.definition` and `.spec.type` fields. Definition should reflect selected type. Additionally `.spec.definition` can directly describe the api verbatim or can reference remote or service provided dynamic API definitions through supported [substitution modifiers](https://backstage.io/docs/features/software-catalog/descriptor-format#substitutions-in-the-descriptor-format).
+
+### Kind: Resource
+
+This resource type represents any other type of dependency of a `Component`. It can be an object bucket, cluster where the workload is hosted at, etc. Any external dependency that the service relies on.
+
+### Kind: User
+
+Represents individual users.
+
+### Kind: Group
+
+Represents group of users or teams. Composable unit of organization and people structure. Groups should represent individual service owners and contact points.
+
+### Kind: Template
+
+In Backstage this is used for interactive forms. Since all our configuration is declarative, we don't plan on using this resource.
+
+### Kind: Location
+
+Backstage provides this resource type to allow you to extend catalog from other locations. Since all our catalog is co-located in a single place, we don't plan on using this resource.
```

Comment:
We are actually using `Kind: Location` [here](https://github.com/operate-first/apps/blob/master/service-catalog/all.yaml) and so the end of the sentence seems a bit misleading.
```suggestion
Backstage provides this resource type to allow you to extend catalog from other locations. Since our catalog is co-located in a single place, we don't plan on using this resource to reference locations other then the repository that defines the catalog.
```